### PR TITLE
Adds optional SNS notification containing results summary.

### LINKED ISSUE #33: Notify the webapp when done
Ref https://github.com/sul-dlss/dlme/issues/410

We need to provide status, any errors and the s3 url.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #33

### CODE REVIEW

**Comment by justinlittman** [comment]
Still needs tests.

**Comment by justinlittman** [comment]
Yippee!
screen shot 2019-02-20 at 3 34 33 pm

File: lib/cli.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@
 require 'json'
 require 'thor'
 require 'transformer'
+require 'record_counter'
+require 'Date'
```
Comment (jcoyne) [inline lib/cli.rb]: this should be `date`.  See the example: https://ruby-doc.org/stdlib-2.3.3/libdoc/date/rdoc/DateTime.html

File: lib/transformer.rb [hunk 2]
```diff
@@ -74,12 +74,12 @@ def initialize(input_filepath:, config_filepaths:, settings: {})
...
     end
 
-    # Transform a stream into a new representation, using Traject
+    # Transform a stream into a new representation, using Traject, and returns the number of transformed records.
     def transform
       transformer.process(File.open(input_filepath, 'r'))
     rescue RuntimeError => e
       warn "[ERROR] #{e.message}"
-      exit(1)
+      raise
```
Comment (jcoyne) [inline lib/transformer.rb]: Do we even need this rescue block? I think we can just handle this in `cli.rb`

**Pull request merged**